servicetalk-http-api/src/main/java/io/servicetalk/http/api/Http2SettingsBuilder.java

```diff
@@ -0,0 +1,201 @@
+     */
+    private static final char MAX_HEADER_LIST_SIZE = 0x6;
+    private final Map<Character, Integer> settings;
+
+    /**
+     * Create a new instance.
+     */
+    public Http2SettingsBuilder() {
+        this(8);
+    }
+
+    /**
+     * Create a new instance.
+     * @param initialSize The initial size of the map.
+     */
+    Http2SettingsBuilder(int initialSize) {
+        settings = new HashMap<>(initialSize);
+    }
+
+    /**
+     * Set the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_HEADER_TABLE_SIZE</a>.
+     *
+     * @param value The value.
+     * @return {@code this}.
+     */
+    public Http2SettingsBuilder headerTableSize(int value) {
+        settings.put(HEADER_TABLE_SIZE, value);
+        return this;
+    }
+
+    /**
+     * Set the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_CONCURRENT_STREAMS</a>.
+     *
+     * @param value The value.
+     * @return {@code this}.
+     */
+    public Http2SettingsBuilder maxConcurrentStreams(int value) {
+        settings.put(MAX_CONCURRENT_STREAMS, value);
+        return this;
+    }
+
+    /**
+     * Set the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_INITIAL_WINDOW_SIZE</a>.
+     *
+     * @param value The value.
+     * @return {@code this}.
+     */
+    public Http2SettingsBuilder initialWindowSize(int value) {
+        settings.put(INITIAL_WINDOW_SIZE, value);
+        return this;
+    }
+
+    /**
+     * Set the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_FRAME_SIZE</a>.
+     *
+     * @param value The value.
+     * @return {@code this}.
+     */
+    public Http2SettingsBuilder maxFrameSize(int value) {
+        settings.put(MAX_FRAME_SIZE, value);
+        return this;
+    }
+
+    /**
+     * Set the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_HEADER_LIST_SIZE</a>.
+     *
+     * @param value The value.
+     * @return {@code this}.
+     */
+    public Http2SettingsBuilder maxHeaderListSize(int value) {
+        settings.put(MAX_HEADER_LIST_SIZE, value);
+        return this;
+    }
+
+    /**
+     * Build the {@link Map} that represents
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">HTTP/2 Setting</a>.
+     * @return the {@link Map} that represents
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">HTTP/2 Setting</a>.
+     */
+    public Http2Settings build() {
+        return new DefaultHttp2Settings(settings);
+    }
+
+    private static final class DefaultHttp2Settings implements Http2Settings {
+        private final Map<Character, Integer> settings;
+
+        private DefaultHttp2Settings(final Map<Character, Integer> settings) {
+            this.settings = settings;
```

Comment:
If ctor doesn't make a copy of the `Map`, builder still have reference to the same map and can change values after `build()`. If you remove support for custom settings, consider using pojo without the internal map.

Reply:
good catch, I will copy.